newIDE/app/src/Utils/UpdateDiscordRichPresence.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import { useEffect, useState } from 'react';
+import optionalRequire from './OptionalRequire';
+const electron = optionalRequire('electron');
+const ipc = electron ? electron.ipcRenderer : null;
+
+const updateDiscordRichPresence = (project: ?gdProject) => {
+  if (ipc === null) return;
+  if (!updateDiscordRichPresence.date)
+    updateDiscordRichPresence.date = Date.now();
```

Comment:
nitpicking: in case of something like this that should outlive the function call, you can as well put it outside `updateDiscordRichPresence`:

```
let richPresenceStartTimestamp = 0;

const updateDiscordRichPresence = (...) => 
```

I find it slightly more explicit because you have to write "let", whereas setting a property on a function looks a bit "implicit" and I would rather avoid messing with the builtin `Function` properties.